### DISCUSSION TITLE
How to show file icons in Filter with Other option

### DISCUSSION THREAD

**Discussion by vishalgoyal2472** [microsoft-search/pnp-modern-search#4114] [General]
How to show file icons in Filter with Other option as in default SharePoint Online search

<img width="164" alt="image" src="https://github.com/user-attachments/assets/0f0b0c8a-414c-4e64-a86a-163aea1f36e5">

<img width="167" alt="image" src="https://github.com/user-attachments/assets/d2485fe4-234e-4225-b392-716751b9b42a">

**Comment by kasperbolarsen** [comment]
This might be pretty close, https://github.com/microsoft-search/pnp-modern-search-layouts/tree/main/Refiner%2FFile%20Extensions%20icons

**Comment by vishalgoyal2472** [comment]
thank you for sharing.

It does not have the same capability which Microsoft search have , Microsoft filter type has clubbed similar file type in one and rest file types in other, please see the logic below they are using, is there any plan to launch the similar feature in future release?

_And if I want to contribute in this community by creating this feature how I can start, I have done small works in SPFX but I would love to contribute._



**1. Word:**
OR(FileType:equals("docx"),FileType:equals("doc"),FileType:equals("docm"),FileType:equals("dot"),FileType:equals("nws"),FileType:equals("dotx"))
->KQL format:
FileType:("docx" OR "doc" OR "docm" OR "dot" OR "nws" OR "dotx")

**2. PowerPoint:**
OR(FileType:equals("odp"),FileType:equals("ppt"),FileType:equals("pptm"),FileType:equals("pptx"),FileType:equals("potm"),FileType:equals("potx"),FileType:equals("ppam"),FileType:equals("ppsm"),FileType:equals("ppsx"))

->KQL format:
FileType:("odp" OR "ppt" OR "pptm" OR "pptx" OR "potm" OR "potx" OR "ppam" OR "ppsm")

**3. Excel:**
or(FileType:equals("odc"),FileType:equals("ods"),FileType:equals("xls"),FileType:equals("xlsb"),FileType:equals("xlsm"),FileType:equals("xlsx"),FileType:equals("xltm"),FileType:equals("xltx"),FileType:equals("xlam"))

->KQL format:
FileType:("odc" OR "ods" OR "xls" OR "xlsb" OR "xlsm" OR "xlsx" OR "xltm" OR "xltx" OR "xlam")

**4. Visio:**
or(FileType:equals("vsd"),FileType:equals("vsx"),FileType:equals("vsdx"),FileType:equals("vdw"),FileType:equals("vdx"),FileType:equals("vss"),FileType:equals("vssx"),FileType:equals("vssm"),FileType:equals("vsdm"),FileType:equals("vstx"),FileType:equals("vstm"))

->KQL format:
FileType:("vsd" OR "vsx" OR "vsdx" OR "vdw" OR "vdx" OR "vss" OR "vssx" OR "vssm" OR "vsdm" OR "vstx" OR "vstm")

**5. OneNote:**
FileType:equals("one")  
->format: FileType:"one"

**6. Pdf:**
FileType:equals("pdf") 
->KQL format: FileType:"pdf"

**7. Loop:**
or(FileType:equals("loop"),FileType:equals("fluid"))
->KQL format:
FileType:("loop" OR "fluid")

**8. Photo:**
or(FileType:equals("bmp"),FileType:equals("jpeg"),FileType:equals("png"),FileType:equals("tiff"),FileType:equals("gif"),FileType:equals("rle"),FileType:equals("wmf"),FileType:equals("dib"),FileType:equals("ico"),FileType:equals("wpd"),FileType:equals("odg"))

->KQL format:
FileType:("bmp" OR "jpeg" OR "png" OR "tiff" OR "gif" OR "rle" OR "wmf" OR "dib" OR "ico" OR "wpd" OR "odg")

**9. Web page:**
or(FileType:equals("HTML"),FileType:equals("MHTML"),FileType:equals("ASPX"))

->KQL format:
FileType:("HTML" OR "MHTML" OR "ASPX")

**10. Video:**
->KQL Format:
FileType:("wmv" OR "avi" OR "mpg" OR "asf" OR "mp4" OR "ogg" OR "ogv" OR "webm" OR "mov" OR "g2" OR "3gp" OR "3gpp" OR "m2ts" OR "m4a" OR "mkv" OR "mp4v" OR "mts" OR "ts" OR "wav" OR "wma")

**11. Others(This is very expensive query):**
->KQL format
NOT (FileType:("docx" OR "doc" OR "docm" OR "dot" OR "nws" OR "dotx" OR "odp" OR "ppt" OR "pptm" OR "pptx" OR "potm" OR "potx" OR "ppam" OR "ppsm" OR "odc" OR "ods" OR "xls" OR "xlsb" OR "xlsm" OR "xlsx" OR "xltm" OR "xltx" OR "xlam" OR "vsd" OR "vsx" OR "vsdx" OR "vdw" OR "vdx" OR "vss" OR "vssx" OR "vssm" OR "vsdm" OR "vstx" OR "vstm" OR "one" OR "pdf" OR "loop" OR "fluid" OR "bmp" OR "jpeg" OR "png" OR "tiff" OR "gif" OR "rle" OR "wmf" OR "dib" OR "ico" OR "wpd" OR "odg" OR "HTML" OR "MHTML" OR "ASPX" OR "wmv" OR "avi" OR "mpg" OR "asf" OR "mp4" OR "ogg" OR "ogv" OR "webm" OR "mov" OR "g2" OR "3gp" OR "3gpp" OR "m2ts" OR "m4a" OR "mkv" OR "mp4v" OR "mts" OR "ts" OR "wav" OR "wma"))

**Comment by kasperbolarsen** [comment]
If the current sample doesn't live up to your expectations or the clients
requirements, it will be up to you to create a new layout. We would love to
receive a PR with an updated layout. The current one, linked above, was
something I made by request of my employer, and as far as I know there are
no plans to expand it.

SPFx skills will unfortunately not be much use, as layouts are using
Handlebars, HTML and CSS

Den tors. 21. nov. 2024 kl. 08.00 skrev vishalgoyal2472 <
***@***.***>:

> thank you for sharing.
>
> It does not have the same capability which Microsoft search have ,
> Microsoft filter type has clubbed similar file type in one and rest file
> types in other, please see the logic below they are using, is there any
> plan to launch the similar feature in future release?
>
> *And if I want to contribute in this community by creating this feature
> how I can start, I have done small works in SPFX but I would love to
> contribute.*
>
> *1. Word:*
>
> OR(FileType:equals("docx"),FileType:equals("doc"),FileType:equals("docm"),FileType:equals("dot"),FileType:equals("nws"),FileType:equals("dotx"))
> ->KQL format:
> FileType:("docx" OR "doc" OR "docm" OR "dot" OR "nws" OR "dotx")
>
> *2. PowerPoint:*
>
> OR(FileType:equals("odp"),FileType:equals("ppt"),FileType:equals("pptm"),FileType:equals("pptx"),FileType:equals("potm"),FileType:equals("potx"),FileType:equals("ppam"),FileType:equals("ppsm"),FileType:equals("ppsx"))
>
> ->KQL format:
> FileType:("odp" OR "ppt" OR "pptm" OR "pptx" OR "potm" OR "potx" OR "ppam"
> OR "ppsm")
>
> *3. Excel:*
>
> or(FileType:equals("odc"),FileType:equals("ods"),FileType:equals("xls"),FileType:equals("xlsb"),FileType:equals("xlsm"),FileType:equals("xlsx"),FileType:equals("xltm"),FileType:equals("xltx"),FileType:equals("xlam"))
>
> ->KQL format:
> FileType:("odc" OR "ods" OR "xls" OR "xlsb" OR "xlsm" OR "xlsx" OR "xltm"
> OR "xltx" OR "xlam")
>
> *4. Visio:*
>
> or(FileType:equals("vsd"),FileType:equals("vsx"),FileType:equals("vsdx"),FileType:equals("vdw"),FileType:equals("vdx"),FileType:equals("vss"),FileType:equals("vssx"),FileType:equals("vssm"),FileType:equals("vsdm"),FileType:equals("vstx"),FileType:equals("vstm"))
>
> ->KQL format:
> FileType:("vsd" OR "vsx" OR "vsdx" OR "vdw" OR "vdx" OR "vss" OR "vssx" OR
> "vssm" OR "vsdm" OR "vstx" OR "vstm")
>
> *5. OneNote:*
> FileType:equals("one")
> ->format: FileType:"one"
>
> *6. Pdf:*
> FileType:equals("pdf")
> ->KQL format: FileType:"pdf"
>
> *7. Loop:*
> or(FileType:equals("loop"),FileType:equals("fluid"))
> ->KQL format:
> FileType:("loop" OR "fluid")
>
> *8. Photo:*
>
> or(FileType:equals("bmp"),FileType:equals("jpeg"),FileType:equals("png"),FileType:equals("tiff"),FileType:equals("gif"),FileType:equals("rle"),FileType:equals("wmf"),FileType:equals("dib"),FileType:equals("ico"),FileType:equals("wpd"),FileType:equals("odg"))
>
> ->KQL format:
> FileType:("bmp" OR "jpeg" OR "png" OR "tiff" OR "gif" OR "rle" OR "wmf" OR
> "dib" OR "ico" OR "wpd" OR "odg")
>
> *9. Web page:*
>
> or(FileType:equals("HTML"),FileType:equals("MHTML"),FileType:equals("ASPX"))
>
> ->KQL format:
> FileType:("HTML" OR "MHTML" OR "ASPX")
>
> *10. Video:*
> ->KQL Format:
> FileType:("wmv" OR "avi" OR "mpg" OR "asf" OR "mp4" OR "ogg" OR "ogv" OR
> "webm" OR "mov" OR "g2" OR "3gp" OR "3gpp" OR "m2ts" OR "m4a" OR "mkv" OR
> "mp4v" OR "mts" OR "ts" OR "wav" OR "wma")
>
> *11. Others(This is very expensive query):*
> ->KQL format
> NOT (FileType:("docx" OR "doc" OR "docm" OR "dot" OR "nws" OR "dotx" OR
> "odp" OR "ppt" OR "pptm" OR "pptx" OR "potm" OR "potx" OR "ppam" OR "ppsm"
> OR "odc" OR "ods" OR "xls" OR "xlsb" OR "xlsm" OR "xlsx" OR "xltm" OR
> "xltx" OR "xlam" OR "vsd" OR "vsx" OR "vsdx" OR "vdw" OR "vdx" OR "vss" OR
> "vssx" OR "vssm" OR "vsdm" OR "vstx" OR "vstm" OR "one" OR "pdf" OR "loop"
> OR "fluid" OR "bmp" OR "jpeg" OR "png" OR "tiff" OR "gif" OR "rle" OR "wmf"
> OR "dib" OR "ico" OR "wpd" OR "odg" OR "HTML" OR "MHTML" OR "ASPX" OR "wmv"
> OR "avi" OR "mpg" OR "asf" OR "mp4" OR "ogg" OR "ogv" OR "webm" OR "mov" OR
> "g2" OR "3gp" OR "3gpp" OR "m2ts" OR "m4a" OR "mkv" OR "mp4v" OR "mts" OR
> "ts" OR "wav" OR "wma"))
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft-search/pnp-modern-search/discussions/4114#discussioncomment-11332752>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE5DXIVCKYFEJPEG5Q7QDOD2BWAIJAVCNFSM6AAAAABSEWFHWCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZTGI3TKMQ>
> .
> You are receiving this because you commented.Message ID:
> <microsoft-search/pnp-modern-search/repo-discussions/4114/comments/11332752
> @github.com>
>


-- 
Med venlig hilsen/ Best regards

Kasper Bo Larsen

**Reply by vishalgoyal2472** [reply]
If I need to merge your one in our solution , Do I need to add it in spfx solution I downloaded or it can be configured in Filter webpart itself.

**Reply by vishalgoyal2472** [reply]
I got it, It need to be merged in Filter webpart layout.

Thank you very much!!

**Comment by kasperbolarsen** [comment]
It is super easy to customize the layouts
https://microsoft-search.github.io/pnp-modern-search/create-custom-layouts/

In this case, you can either copy the html file content into the "Edit filters template", replacing the existing one ( you will most likely have to adjust the Slot setting, if the layout is using some Slots that is not the default ones). The alternative, with I recommend, is to save the HTML file on the tenant and just insert the true URL to the file as below:
![image](https://github.com/user-attachments/assets/2a57cd65-b8bf-4e06-a078-fde28f53f17a)

This might be useful too: https://www.m365thinking.com/post/work-with-your-handlebar-based-layouts-in-an-interactive-way

**Reply by vishalgoyal2472** [reply]
I need to show icons in combo box dropdown, your solution is designed for checkbox.

**Comment by kasperbolarsen** [comment]
As far as I know there are no samples available for icons in the text fields within a combobox, and you will have to look into whether that is possible at all with the react control we are using. 
Based on the number of questions and issue you have raised recently, I'll recommend adjusting your requirements or building your own webpart from scratch if the customer requirements are fixed.

**Reply by vishalgoyal2472** [reply]
I tried to modify like below, its showing text '22_testing' but not showing file icon, I tried same pnp-icon at other places and its showing file icon correctly, 
Anything I am missing here?

`<pnp-filtercombobox
                                data-instance-id="{{@root.instanceId}}"
                                data-filter-name="{{filter.filterName}}"
                                data-is-multi="{{filter.isMulti}}"
                                data-show-count="{{filter.showCount}}"
                                data-operator="{{filter.operator}}"
                                data-theme-variant="{{JSONstringify @root.theme}}">
                            {{#each filter.values}}
                                <option value="{{value}}" data-selected="{{selected}}" data-disabled="{{disabled}}" data-count="{{count}}">
                                    {{name}}22_testing
                                        <pnp-iconfile data-extension="docx" data-is-container="false"  data-size="32"></pnp-iconfile>
                                            <pnp-iconfile data-extension="{{name}}" data-is-container="false" data-size="32"></pnp-iconfile>
                                        </option>
                            {{/each}}
                            </pnp-filtercombobox>    `
                            
                            
<img width="161" alt="image" src="https://github.com/user-attachments/assets/2173d92e-30d8-4aa8-bd7c-7ea8752bded7">

**Comment by vishalgoyal2472** [comment]
I want to build this feature to add file icons in Multidrop down.

Do I need to implement it in SPFX or Handlebars?

How can I start and add my contribution to the community?

**Reply by vishalgoyal2472** [reply]
Could someone please respond?

**Reply by wobba** [reply]
A SPFx special template seems easier I think as you need to collate doc/docx/dot/dotx etc into one icon for Word as an example.
